morphir/ir/test/src/morphir/ir/DistributionSpec.scala

```diff
@@ -0,0 +1,42 @@
+package morphir.ir
+
+import io.circe.{ Decoder, Json, parser }
+import io.circe.Decoder.Result
+import morphir.ir.Distribution.Distribution
+import morphir.ir.distribution.Codec
+import zio.test.Assertion.equalTo
+import zio.test._
+
+import scala.io.{ BufferedSource, Source }
+
+object DistributionSpec extends DefaultRunnableSpec {
+  val distributionDecoder: Decoder[Distribution] = (c: io.circe.HCursor) =>
+    c.downField("distribution").as(Codec.decodeDistribution)
+  val bufferResource: BufferedSource = Source.fromFile(
+    "morphir/ir/test/resources/morphir/ir/morphir-ir.json"
+  )
+  val irContent: String = bufferResource.mkString
+  bufferResource.close()
+
+  override def spec: ZSpec[_root_.zio.test.environment.TestEnvironment, Any] = suite("DistributionSpec")(
+    suite("Comparism of Generated IR Json vs Encoded Distribution to IR Json")(
+      test("Decode Distribution from IRJson + Re-Encode Distribution + Compare ") {
+        val irJson                                   = parser.parse(irContent).getOrElse(Json.Null)
+        val distributionResult: Result[Distribution] = distributionDecoder.decodeJson(irJson)
+        distributionResult match {
+          case Left(_) => assert(true)(equalTo(false)) // cant find an intentional fail
```

Comment:
I've been searching for a better approach but it seems like there isn't any. I don't even understand why we have ZIO mixed in here. ZIO is for side-effects and there's no side-effects here. Can we remove the ZIO aspect and just use plain ScalaTest for this? I think the sdk subproject does that, so that can be used as a reference.